src/Uno.UI/UI/Xaml/Controls/Popup/Popup.Base.cs

```diff
@@ -157,7 +165,7 @@ private void UpdateDataContext(DependencyPropertyChangedEventArgs e)
 					// Child already has locally set DataContext, we shouldn't overwrite it.
 					_childHasOwnDataContext = true;
 				}
-				else
+				else if(PropagatesDataContextToChild)
```

Comment:
Is there a problem if we skipped the whole `UpdateDataContext` on this condition?

Reply:
There isn't, let's do that for now :)